tf_quant_finance/math/pde/fd_solvers.py

```diff
@@ -81,63 +81,76 @@ def solve_backward(start_time,
 
   ### Example. European call option pricing.
   ```python
+  import tensorflow as tf
+  import tf_quant_finance as tff
+  pde = tff.math.pde
+
   num_equations = 2  # Number of PDE
   num_grid_points = 1024  # Number of grid points
-  dtype = np.float64
+  dtype = tf.float64
+
   # Build a log-uniform grid
   s_min = 0.01
   s_max = 300.
-  grid = uniform_grid(minimums=[s_min], maximums=[s_max],
-                      sizes=[num_grid_points],
-                      dtype=dtype)
+  grid = pde.grids.uniform_grid(minimums=[s_min],
+                                maximums=[s_max],
+                                sizes=[num_grid_points],
+                                dtype=dtype)
+
   # Specify volatilities and interest rates for the options
+  strike = tf.constant([[50], [100]], dtype)
   volatility = tf.constant([[0.3], [0.15]], dtype)
   rate = tf.constant([[0.01], [0.03]], dtype)
   expiry = 1.0
-  strike = tf.constant([[50], [100]], dtype)
+
+  # For batching multiple PDEs, we need to stack the grid values
+  # so that final_values[i] is the grid for the ith strike.
+  # We will use `tf.meshgrid` to quickly do this.
+  s = grid[0]
+  strike_mesh, s_mesh = tf.meshgrid(strike, s, indexing='ij')
+  final_value_grid = tf.nn.relu(s_mesh - strike_mesh)
 
   # Define parabolic equation coefficients. In this case the coefficients
   # can be computed exactly but the same functions as below can be used to
   # get approximate values for general case.
+  # We again need to use `tf.meshgrid` to batch the coefficients.
   def second_order_coeff_fn(t, grid):
-    del t
-    x = grid[0]
-    return [[tf.square(volatility) * tf.square(x) / 2]]
+    s = grid[0]
+    vol_mesh, s_mesh = tf.meshgrid(volatility, s, indexing='ij')
+    return [[vol_mesh**2 * s_mesh**2 / 2]]
 
   def first_order_coeff_fn(t, grid):
-    del t
-    x = grid[0]
-    return [rate * x]
+    s = grid[0]
+    rate_mesh, s_mesh = tf.meshgrid(rate, s, indexing='ij')
+    return [rate_mesh * s_mesh]
 
   def zeroth_order_coeff_fn(t, grid):
-    del t, grid
-    return -rate
-
-  @boundary_conditions.dirichlet
-  def lower_boundary_fn(t, x):
-    del t, x
-    return dtype([0.0, 0.0])
-
-  @boundary_conditions.dirichlet
-  def upper_boundary_fn(t, x):
-    return tf.squeeze(x - strike * tf.exp(-rate * (expiry - t)))
-
-  final_values = tf.nn.relu(grid - strike)
-  # Broadcast to the shape of value dimension, if necessary.
-  final_values += tf.zeros([num_equations, num_grid_points],
-                           dtype=dtype)
+    s = grid[0]
+    rate_mesh, s_mesh = tf.meshgrid(rate, s, indexing='ij')
+    return -rate_mesh
+
+  @pde.boundary_conditions.dirichlet
+  def lower_boundary_fn(t, grid):
+    return tf.constant(0.0, dtype=dtype)
```

Comment:
since the function does not depend in `t, grid`, we explicitly state this `del t, grid`. Please apply where necessary

Reply:
`del t, grid` is required by style guide, but users don't have to follow it, and this deletion may look confusing to those who don't. So I'd say `del t, grid` is not necessary here.

Reply:
del t, grid is required by style guide, but users don't have to follow it, and this deletion may look confusing to those who don't. So I'd say del t, grid is not necessary here.

Reply:
I've added them in for explicit compliance. I hope that's alright.